tests/default.nix

```diff
@@ -214,8 +214,14 @@ import nmtSrc {
             dir:
             lib.pipe dir [
               builtins.readDir
-              (lib.filterAttrs (_path: kind: kind == "directory"))
-              (lib.mapAttrsToList (path: _kind: lib.path.append dir path))
+              (lib.mapAttrsToList (
+                path: kind:
+                if kind == "directory" || (kind == "regular" && lib.hasSuffix ".nix" path) then
+                  lib.path.append dir path
+                else
+                  null
+              ))
+              (lib.filter (x: x != null && !lib.hasPrefix "_" (baseNameOf (toString x))))
```

Comment:
Do we really care to support it here? We've always done folders for tests.

Reply:
If we dont need to, let's not do it. Simpler is better and tests are less visible to the users.
The extra complexity is worth for autoimporting module because it is more backwards compatible.

Reply:
I'll pull it out when I rebase after merging https://github.com/nix-community/home-manager/pull/7303 since she already rebased on the current way.